### PR TITLE
1869 error handling

### PR DESCRIPTION
**Pull request by IagoLast**
#1869
----

- Client throws cartoErrors wrapped from engine errors.
- `.addLayer` and `.addDataview` return a promise, if those promises are rejected the parameter is a CartoError
- Add an error example with a code snippet and description.
- Update carto.events doc, use a custom namespace with constants instead enum

[https://gyazo.com/cb4c6c96d4c9f31c7e1dc87e6f2ff5dd](https://gyazo.com/cb4c6c96d4c9f31c7e1dc87e6f2ff5dd)

### CODE REVIEW

File: src/api/v4/error.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
+ * Build a cartoError from a generic error
+ * @constructor
+ * 
+ * @return {CartoError} A well formed object representing the error.
+ */
+function CartoError (error) {
+  if (!error) {
```
Comment (alonsogarciapablo) [inline src/api/v4/error.js]: I think we can get rid of this if (we're also assigning 'unexpected error' to `this.message` in line #20)

File: src/api/v4/error.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  return error && error.errors_with_context;
+}
+
+module.exports = CartoError;
+
+/**
+ * 
+ * Represents an error in the carto library.
+ * 
+ * Some actions like adding a layer to a map will trigger a **reload cycle**
```
Comment (alonsogarciapablo) [inline src/api/v4/error.js]: "reload cycle" is not very clear to me. I'd try to find something more meaningful for developers.

Comment (IagoLast) [reply src/api/v4/error.js]: umm, any suggestion?

I would try to explain how carto.js works (including the reload-cycle) to developers in order to make it easier to understand

File: examples/v4/category-dataview.html [hunk 3]
```diff
@@ -35,15 +35,10 @@
...
-      // var client = new carto.Client({
-      //   apiKey: '84fdbd587e4a942510270a48e843b4c1baa11e18',
-      //   serverUrl: 'https://{user}.carto.com:443',
-      //   username: 'cartojs-test'
-      // });
 
       var client = new carto.Client({
         apiKey: '4bf183ec5867bd5f9eb59d4d33693fa4fb4283bf',
-        serverUrl: 'http://{user}.localhost.lan:8181',
+        serverUrl: 'http://cdb.localhost.lan:8181',
```
Comment (IagoLast) [inline examples/v4/category-dataview.html]: use carto-js-test

**Pull request merged**